core/src/main/java/hudson/model/ItemGroup.java

```diff
@@ -55,6 +57,21 @@
      */
     Collection<T> getItems();
 
+    /**
+     * Gets all the items in this collection in a read-only view
+     * that matches supplied Predicate
+     * @since TODO
+     */
+     default Collection<T> getItems(Predicate<T> pred) {
+         List<T> filteredItems = new ArrayList<>();
```

Comment:
Not important at all, but maybe using stream + predicate?

Reply:
Implemented